### ISSUE TITLE
Regenerate OSM custom tilesets [incl hillshading & contours?]

### ISSUE THREAD

**Issue by m-anish** [iiab/iiab#1423]
Experiment with having our own tileserver to do our own tilesets including hillshading and contours, for future deployments.

**Comment by holta** [comment]
@georgejhunt is helping @m-anish with Braddock's server here.

Refs: #877 #1170

**Comment by holta** [comment]
Context: we're trying to use vector tiles as much as possible — as described in http://FAQ.IIAB.IO here:

http://wiki.laptop.org/go/IIAB/FAQ#How_do_I_add_zoomable_maps_for_my_region.3F
https://github.com/iiab/iiab-factory/blob/master/content/vector-tiles/README.md
http://download.iiab.io/content/OSM/vector-tiles/
https://openmaptiles.com/downloads/

**Comment by holta** [comment]
fyi @georgejhunt is researching elegant tile generation &mdash; different approaches/algorithms to deliver vivid-yet-not-overcrowded vector tiles &mdash; building on his/our 2018 work (["How do I add zoomable maps for my region?"](http://wiki.laptop.org/go/IIAB/FAQ#How_do_I_add_zoomable_maps_for_my_region.3F)).

This is related to #1192 from October 2018, but represents a much larger challenge around generating quality visualizations & illustration in general (Edward Tufte etc!)

Umbrella Ticket of OSM Priorities arising from 2018: https://github.com/iiab/iiab/issues/877#issuecomment-405935272

**Comment by holta** [comment]
@m-anish I'm deferring this hill-shading, contour line drawing & similar work to the IIAB 7.1 milestone, assuming it's not converging in coming weeks in time for IIAB 7.0?

(Or if I'm wrong, just LMK!)

**Comment by holta** [comment]
@mir123 warns that this might risk becoming a Sisyphean (overwhelmingly large) task that might bog us down — and never quite be completed well/sustainably.

Certainly: many/most OpenStreetMap offshoot/derivative projects use known tilesets like https://OpenMapTiles.org as there's a massive cost to reinventing the wheel, i.e. particularly when an huge/subtle/evolving ecosystem of scraping, packaging (and publishing if not archiving!) is at stake.

**Comment by mir123** [comment]
It is possible to generate your own tiles for sure, but it's indeed a significant undertaking, especially for large countries or subcontinent-size bounding boxes. I had success with using https://github.com/openmaptiles/openmaptiles, about 10 hours processing time for Central America and that's without hillshading and contours. I think this might be outside of the project's scope, but within the realm of the possible for those use cases that require up to date maps

**Comment by m-anish** [comment]
> @m-anish I'm deferring this hill-shading, contour line drawing & similar work to the IIAB 7.1 milestone, assuming it's not converging in coming weeks in time for IIAB 7.0?
> 
> (Or if I'm wrong, just LMK!)

Right. Its not happening anytime soon. But it did happen in the past (I did it) and it will happen again in the future, but don't make time-bound plans around it. That's the best I can offer at the moment. Sorry.

**Comment by m-anish** [comment]
And, with help from the OSM community it took around a week using spinning hard disks to generate high res maps including contours and hillshade for India, so I think it's certainly worthwhile, especially in mountainous regions where maps without hillshade really dont mean anything.

**Comment by mir123** [comment]
Anish, do you know the procedure for generating the hillshades and contours
and to include them in the mbtiles?

**Comment by m-anish** [comment]
I dont know how to do it in mbtiles, but I did know how to do it in metatiles. It needs time and lots of bandwidth and resources, but it is possible.
